int16 num1
int16 num2
int16 num3